File: src/server/src/test/base_test.py
import unittest
from abc import ABC

import velvet_dawn
from velvet_dawn.config import Config
from velvet_dawn.dao import app, db


class BaseTest(unittest.TestCase, ABC):

    def get_test_config(self):
        config = Config()
        config.datapacks = ['__testing__', 'civil-war']
        return config

    def setup_game(self):
        with app.app_context():
            config = self.get_test_config()
            config.seed = 0
            velvet_dawn.datapacks.init(config)
            velvet_dawn.map.creation.new(config)
            velvet_dawn.players.join("player1", "")
            velvet_dawn.players.join("player2", "")
            velvet_dawn.game.setup.update_setup("testing:commander", 1)
            velvet_dawn.game.setup.update_setup("civil-war:commander", 1)
            velvet_dawn.game.setup.update_setup("civil-war:cavalry", 1)
            velvet_dawn.game.phase.start_setup_phase(config)
            velvet_dawn.game.setup.place_entity("player1", "testing:commander", 15, 0, config)
            velvet_dawn.game.setup.place_entity("player1", "civil-war:cavalry", 14, 0, config)
            instance = velvet_dawn.game.setup.place_entity("player2", "testing:commander", 15, config.map_height - 1, config)
            velvet_dawn.game.phase.start_game_phase(config)

            velvet_dawn.db.units.move(instance, 15, 2)

    def get_config(self):
        config = Config()
        config.datapacks = ['__testing__']
        config.map_height = 11
        config.map_width = 7
        config.spawning.width_multiplier = 1
        config.spawning.neighbours_multiplier = 1
        config.spawning.width_addition = 1
        return config

    def prepare_game(self):
        with app.app_context():
            config = self.get_config()
            config.seed = 0
            velvet_dawn.datapacks.init(config)
            velvet_dawn.map.creation.new(config)
            velvet_dawn.players.join("player1", "")
            velvet_dawn.players.join("player2", "")
            velvet_dawn.game.setup.update_setup("testing:commander", 1)
            velvet_dawn.game.setup.update_setup("testing:upgradable", 1)
            velvet_dawn.game.setup.update_setup("testing:abilitied", 1)
            velvet_dawn.game.phase.start_setup_phase(config)
            velvet_dawn.game.setup.place_entity("player1", "testing:commander", 5, 0, config)
            velvet_dawn.game.setup.place_entity("player1", "testing:upgradable", 4, 0, config)
            velvet_dawn.game.setup.place_entity("player1", "testing:abilitied", 6, 0, config)
            instance = velvet_dawn.game.setup.place_entity("player2", "testing:commander", 5, config.map_height - 1,
                                                config)
            velvet_dawn.game.phase.start_game_phase(config)

            velvet_dawn.db.units.move(instance, 5, 2)

    def setUp(self) -> None:
        velvet_dawn.db.instance.clear()
        with app.app_context():
            db.drop_all()
            db.create_all()

File: src/jserver/src/test/BaseTest.java
package test;

import velvetdawn.VelvetDawn;
import velvetdawn.models.Coordinate;
import velvetdawn.models.config.Config;

import java.util.List;

public abstract class BaseTest {

    public Config getTestConfig() {
        Config config = new Config();
        config.datapacks = List.of("__testing__", "civil-war");
        return config;
    }

    public Config getConfig() {
        Config config = new Config();
        config.datapacks = List.of("__testing__");
        config.map.borderRadius = 5;
        config.map.seed = 0;
        config.spawn.spawnRadiusMultiplier = 1;
        config.spawn.baseSpawnRadius = 1;
        return config;
    }

    public VelvetDawn setupGame() throws Exception {
        Config config = this.getTestConfig();
        VelvetDawn velvetDawn = new VelvetDawn(config);
        velvetDawn.init();

        var p1 = velvetDawn.players.join("player1", "null");
        var p2 = velvetDawn.players.join("player2", "null");

        velvetDawn.game.setup.updateSetup("testing:commander", 1);
        velvetDawn.game.setup.updateSetup("civil-war:commander", 1);
        velvetDawn.game.setup.updateSetup("civil-war:cavalry", 1);
        velvetDawn.game.startSetupPhase();
        velvetDawn.game.setup.placeEntity(p1, "testing:commander", new Coordinate(15, 0));
        velvetDawn.game.setup.placeEntity(p1, "civil-war:cavalry", new Coordinate(14, 0));
        var instance = velvetDawn.game.setup.placeEntity(p2, "testing:commander", new Coordinate(15, config.map.borderRadius));
        velvetDawn.game.startGamePhase();

        velvetDawn.entities.move(instance, new Coordinate(15, 2));

        return velvetDawn;
    }

    public VelvetDawn prepareGame() throws Exception {
        Config config = this.getConfig();
        VelvetDawn velvetDawn = new VelvetDawn(config);
        velvetDawn.init();

        var p1 = velvetDawn.players.join("player1", "null");
        var p2 = velvetDawn.players.join("player2", "null");

        velvetDawn.game.setup.updateSetup("testing:commander", 1);
        velvetDawn.game.setup.updateSetup("testing:upgradable", 1);
        velvetDawn.game.setup.updateSetup("testing:abilitied", 1);
        velvetDawn.game.startSetupPhase();
        velvetDawn.game.setup.placeEntity(p1, "testing:commander", new Coordinate(0, -config.map.borderRadius));
        velvetDawn.game.setup.placeEntity(p1, "testing:upgradable", new Coordinate(-1, -config.map.borderRadius));
        velvetDawn.game.setup.placeEntity(p1, "testing:abilitied", new Coordinate(1, -config.map.borderRadius));
        var instance = velvetDawn.game.setup.placeEntity(p2, "testing:commander", new Coordinate(0, config.map.borderRadius));
        velvetDawn.game.startGamePhase();

        velvetDawn.entities.move(instance, new Coordinate(5, 2));

        return velvetDawn;
    }
}
